.github/workflows/ci-cd.yml

```diff
@@ -153,3 +192,98 @@ jobs:
+    permissions: write-all
+    runs-on: ubuntu-latest
+    needs: build_and_push
+    outputs:
+      RELEASE_ID: ${{ steps.prerelease.outputs.RELEASE_ID }}
+    env:
+      TAG: ${{ needs.build_and_push.outputs.TAG }}
+    steps:
+    - name: Create prerelease
+      id: prerelease
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        export RELEASE_ID=$( jq -r '.id' <<< $(
+        gh api \
+        --method POST \
+        -H "Accept: application/vnd.github+json" \
+        -H "X-GitHub-Api-Version: 2022-11-28" \
+        /repos/AxisCommunications/acap-runtime/releases \
+        -f tag_name='${{ env.TAG }}' \
+        -f target_commitish='main' \
+        -f name='${{ env.TAG }}' \
+        -f body='Prerelease for ${{ env.TAG }}' \
+        -F draft=false \
+        -F prerelease=true \
+        -F generate_release_notes=false
+          ))
+        delimiter="$(openssl rand -hex 8)"
+        echo "RELEASE_ID<<${delimiter}" >> ${GITHUB_OUTPUT}
+        echo "${RELEASE_ID}" >> ${GITHUB_OUTPUT}
+        echo "${delimiter}" >> ${GITHUB_OUTPUT}
+        echo "RELEASE_ID ${RELEASE_ID}"
+
+        if [[ -z ${RELEASE_ID} ]]
+        then
+          echo "Create prerelease was not successful"
+          exit 1
+        else
+          echo "Create prerelease was successful, with id ${RELEASE_ID}"
+        fi
+
+  download-and-upload-artifacts:
+    permissions: write-all
+    runs-on: ubuntu-latest
+    needs: [create_prerelease, build_and_push]
+    strategy:
+      matrix:
+        arch: [armv7hf, aarch64]
+    env:
+      RELEASE_ID: ${{ needs.create_prerelease.outputs.RELEASE_ID }}
+      EAP_FILE_AARCH64: ${{ needs.build_and_push.outputs.EAP_FILE_AARCH64 }}
+      EAP_FILE_ARMV7HF: ${{ needs.build_and_push.outputs.EAP_FILE_ARMV7HF }}
+    steps:
+    - name: Get EAP file name
+      id: full_eap_name
+      run: |
+        if [ ${{ matrix.arch }} = armv7hf ]
+        then
+          echo "EAP_FILE=${{ needs.build_and_push.outputs.EAP_FILE_ARMV7HF }}" >> $GITHUB_ENV
+        elif [ ${{ matrix.arch }} = aarch64 ]
+        then
+          echo "EAP_FILE=${{ needs.build_and_push.outputs.EAP_FILE_AARCH64 }}" >> $GITHUB_ENV
+        else
+          echo "::error::Non valid architecture '${{ matrix.arch }}' encountered"
+        fi
+    - name: Download artifacts
+      uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a
+      with:
+        name: ${{ env.EAP_FILE }}
+        path: ./
+    - name: Upload file to GitHub release
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        RESPONSE=$(curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+          -H "Accept: application/vnd.github.manifold-preview" \
+          -H "Content-Type: application/zip" \
+          --data-binary @${{ env.EAP_FILE }} \
+          "https://uploads.github.com/repos/AxisCommunications/acap-runtime/releases/${{env.RELEASE_ID}}/assets?name=${{ env.EAP_FILE }}" \
```

Comment:
`AxisCommunication/acap-runtime`can be replaced by `${{ vars.GITHUB_REPOSITORY }}`